src/com/google/javascript/jscomp/ProcessCommonJSModules.java

```diff
@@ -222,6 +289,48 @@ private void visitScript(NodeTraversal t, Node script) {
       compiler.reportCodeChange();
     }
 
+    /**
+     * Rewrite CommonJS part of UMD pattern. Pull out then branch,
+     * which includes "module.exports" or "exports" statement
+     * and remove the else branch.
+     */
+    private void visitCommonJSIfCondition(Node n) {
+      Node p = n.getParent();
+      if (p != null) {
+        // pull out then branch
+        pullOutBranch(n.getChildAtIndex(1));
+      }
+    }
+
+    /**
+     * Rewrite AMD part UMD pattern. Pull out else branch since it could
+     * check for CommonJS and remove the then branch.
+     */
+    private void visitAMDIfCondition(Node n) {
+      Node p = n.getParent();
+      if (p != null) {
+        if (n.getChildCount() == 3) {
+          // pull out else branch
+          pullOutBranch(n.getChildAtIndex(2));
+        } else {
+          p.removeChild(n);
+        }
+      }
+    }
+
+    private void pullOutBranch(Node branch) {
+        Node ifNode = branch.getParent();
+        Node p = ifNode.getParent();
+        Node newNode = branch;
+        // remove redundant block node
```

Comment:
Is this required? If not I wouldn't do it, it'll be done by the optimizer later anyway, won't it?